webrtc/RTCPeerConnection-setRemoteDescription.html

```diff
@@ -44,6 +46,42 @@
       assert_unreached('Error ' + e.name + ': ' + e.message);
     }));
   }, 'Triggers ontrack when called with a remote description and the MSID of the stream is is parsed.');
+
+  promise_test(t => {
+    const pc = new RTCPeerConnection();
+    return promise_rejects(t, new RTCError(),
+      pc.setRemoteDescription({
+        // valid SDP type
+        type: 'offer',
+        // malformed SDP description
+        sdp: 'bogus'
+      }));
+  }, 'Malformed SDP description should be rejected with RTCError');
+
+  promise_test(t => {
+    const pc = new RTCPeerConnection();
+    return promise_rejects(t, new TypeError(),
+      pc.setRemoteDescription({
+        // invalid enum value is caught at IDL level before
+        // method is executed
+        type: 'bogus',
+        // bogus SDP should never be validated before type
+        sdp: 'bogus'
+      }));
+  }, 'Invalid SDP type should be rejected with TypeError');
+
+  promise_test(t => {
+    const pc = new RTCPeerConnection();
+
+    return promise_rejects(t, new InvalidStateError(),
```

Comment:
`assert_throws` (used by `promise_rejects`) is a bit funny. This should be 'InvalidStateError', because it's the name of a DOMException and not its own object. `new TypeError` and `new RTCError` are correct though, because that's supposed to work.